File: hdu/1002.java
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: netcan
////Nickname: Netcan
////Run ID: 
////Submit time: 2015-05-30 20:40:15
////Compiler: Java
//////////////////////////////////////////////////////
////Problem ID: 1002
////Problem Title: 
////Run result: Accept
////Run time:624MS
////Run memory:25784KB
//////////////////System Comment End//////////////////
/*************************************************************************
	> File Name: j1002.java
	  > Author: Netcan
	  > Blog: http://www.netcan.xyz
	  > Mail: 1469709759@qq.com
	  > Created Time: Sat 30 May 2015 08:21:08 PM CST
 ************************************************************************/

import java.io.*;
import java.math.*;
import java.util.*;
import java.text.*;

public class Main {
    public static void main(String[] args) {
		Scanner cin = new Scanner(new BufferedInputStream(System.in));
		int T;
		T=cin.nextInt();
		BigInteger a, b;
		int Case = 1;
		for(int T_T=0; T_T<T; ++T_T) {
			a = cin.nextBigInteger();
			b = cin.nextBigInteger();
			System.out.println("Case " + (Case++) + ":");
			System.out.println(a+" + "+b+" = "+a.add(b));
			if(T_T!=T-1)
				System.out.println();
		}

	}
}

File: HDOJ/1002.cpp
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: netcan
////Nickname: Netcan
////Run ID: 
////Submit time: 2014-11-07 15:38:11
////Compiler: GUN C++
//////////////////////////////////////////////////////
////Problem ID: 1002
////Problem Title: 
////Run result: Accept
////Run time:0MS
////Run memory:204KB
//////////////////System Comment End//////////////////
#include<stdio.h>
#include<string.h>
int main()
{
	char a[1000],b[1000],c[1001];
	int i,j=1,p=0,n,n1,n2;
	scanf("%d",&n);
    while(n)
	{
		scanf("%s %s",a,b);
		printf("Case %d:\n",j);
		printf("%s + %s = ",a,b);
		n1=strlen(a)-1;
		n2=strlen(b)-1;
		for(i=0;n1>=0||n2>=0;i++,n1--,n2--)
		{
			if(n1>=0&&n2>=0){c[i]=a[n1]+b[n2]-'0'+p;}
			if(n1>=0&&n2<0){c[i]=a[n1]+p;}
			if(n1<0&&n2>=0){c[i]=b[n2]+p;}
			p=0;
			if(c[i]>'9'){c[i]=c[i]-10;p=1;}
		}
		if(p==1){printf("%d",p);}
		while(i--)
		{printf("%c",c[i]);}
		j++;
		if(n!=1){printf("\n\n");}
		else {printf("\n");}
		n--;
	}
}